File: simulator/effects/square_aoe.py
from ..battle_map import Map


class SquareAoe:

    def __init__(self, origin, length):
        self.origin = origin
        self.length = length
        self.affected_coords = Map.get().get_coords_affected_by_square_aoe(self.origin, self.length)

    def get_affected_coords(self):
        """
        Gets coordinates of grid squares affected by a square effect originating at bottom left corner of a square at origin coordinates.
        :return: affected coordinates as a np.array of nx2 where n is the number of coordinates returned
        """
        return self.affected_coords

File: c++/effects/square_aoe.cpp
#include "effects/square_aoe.hpp"
#include "core/geometry.hpp"
#include "core/battle_map.hpp"

namespace enc
{
  SquareAoe::SquareAoe(const Coord &origin, int length) : _origin(origin), _length(length)
  {
    _affectedCoords = getCoordsAffectedBySquareAoE(_origin, _length, BattleMap::getInstance().getGridSize());
  }

  const CoordVector &SquareAoe::getAffectedCoords() const { return _affectedCoords; }
}
